plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java

```diff
@@ -200,21 +201,40 @@
 
         this.storageClass = STORAGE_CLASS_SETTING.get(metadata.settings());
         this.cannedACL = CANNED_ACL_SETTING.get(metadata.settings());
+
+        if (CLIENT_NAME.exists(metadata.settings()) && S3ClientSettings.checkDeprecatedCredentials(metadata.settings())) {
+            logger.warn("ignoring use of named client [" + metadata.name() + "] as insecure credentials were specified");
```

Comment:
Thanks for adding this warning. Since ` metadata.name() ` refers to the repository name, we could maybe change this to: "ignoring use of named client for repository ["

Reply:
I pushed 730e07c9383f026bb10cb5b4ba056b9011add0a7.